### PR TITLE
executor: add partition table testcase for historical stats

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

add partition table testcase for historical stats

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/dump.go [hunk 1]
```diff
@@ -173,9 +175,8 @@ func (h *Handle) DumpHistoricalStatsBySnapshot(dbName string, tableInfo *model.T
 	if isDynamicMode {
 		tbl, err := h.tableHistoricalStatsToJSON(tableInfo.ID, snapshot)
 		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		if tbl != nil {
+			logutil.BgLogger().Warn("dump global historical stats failed", zap.Int64("table-id", tableInfo.ID))
```
Comment (qw4990) [inline statistics/handle/dump.go:178]: Can we record the table name as well?

File: domain/historical_stats.go [hunk 2]
```diff
@@ -77,5 +77,10 @@ func (w *HistoricalStatsWorker) DumpHistoricalStats(tableID int64, statsHandle *
 
 // GetOneHistoricalStatsTable gets one tableID from channel, only used for test
 func (w *HistoricalStatsWorker) GetOneHistoricalStatsTable() int64 {
```
Comment (qw4990) [inline domain/historical_stats.go:79]: Is this function only used in testing? If not, I'm afraid the `-1` can cause some panic.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**